toolchain/lex/tokenized_buffer.cpp

```diff
@@ -49,21 +55,29 @@ auto VariantMatch(V&& v, Fs&&... fs) -> decltype(auto) {
   return std::visit(Overload{std::forward<Fs&&>(fs)...}, std::forward<V&&>(v));
 }
 
-#if __x86_64__
-#define CARBON_USE_SIMD 1
-
+#if CARBON_USE_SIMD
+namespace {
+#if __ARM_NEON
+using SIMDMaskT = uint8x16_t;
+#elif __x86_64__
+using SIMDMaskT = __m128i;
+#else
+#error "Unsupported SIMD architecture!"
+#endif
+using SIMDMaskArrayT = std::array<SIMDMaskT, sizeof(SIMDMaskT) + 1>;
+}  // namespace
 // A table of masks to include 0-16 bytes of an SSE register.
-// TODO: Make this constexpr to avoid dynamic initialization.
-static const std::array<__m128i, sizeof(__m128i) + 1> prefix_masks = [] {
-  std::array<__m128i, sizeof(__m128i) + 1> masks = {};
-  for (auto [i, mask] : llvm::enumerate(masks)) {
-    memset(&mask, 0xFF, i);
+static constexpr SIMDMaskArrayT PrefixMasks = []() constexpr {
+  SIMDMaskArrayT masks = {};
+  for (int i = 1; i < static_cast<int>(masks.size()); ++i) {
+    // The SIMD types and constexpr require a C-style cast.
+    // NOLINTNEXTLINE(google-readability-casting)
+    masks[i] = (SIMDMaskT)((static_cast<unsigned __int128>(0) - 1) >>
```

Comment:
```suggestion
    masks[i] = (SIMDMaskT)(std::numeric_limits<unsigned __int128>::max() >>
```

Is numeric limits going to give you what you want?

Reply:
Eh, I don't find this as easy to read, but sure.